lib/class-wp-theme-json-gutenberg.php

```diff
@@ -3228,6 +3338,35 @@ public function get_raw_data() {
 		return $this->theme_json;
 	}
 
+	/**
+	 * Converts block styles registered through the `WP_Block_Styles_Registry`
+	 * with a style object, into theme.json format.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return array Styles configuration adhering to the theme.json schema.
+	 */
+	public static function get_from_block_styles_registry() {
```

Comment:
Block style variations will also now support registering them through the style registry with a style object. This function collects those style objects and returns them all as theme json data that can be merged into theme.json objects from other objects. See the changes to `WP_Theme_JSON_Resolver_Gutenberg` for more details.